allauth/socialaccount/providers/windowslive/provider.py

```diff
@@ -7,6 +7,9 @@
 
 class WindowsLiveAccount(ProviderAccount):
     def to_str(self):
+        email = self.account.extra_data.get("emails", {}).get("preferred")
+        if email:
+            return email
         name = "{0} {1}".format(
             self.account.extra_data.get("first_name", ""),
             self.account.extra_data.get("last_name", ""),
```

Comment:
This `name = "{0} ..."` is not WindowsLive specific, we could rely on the base class here.

Reply:
The base `to_str` does not look at `first_name` and `last_name`, are you saying that it should?